dfttk/debye.py

```diff
@@ -0,0 +1,348 @@
+import numpy as np
+import plotly.graph_objects as go
+import os
+
+from scipy import constants
+from scipy.special import bernoulli
+from scipy.special import gamma
+
+from dfttk import eos_fit
+from dfttk.data_extraction import extract_average_mass
+from dfttk.aggregate_extraction import extract_configuration_data
+from dfttk.qha_yphon import plot_format
+
+
+
+# A is the ____ constant of th Debye-Gruneisen model in units of
+    # A = (8 * constants.pi**2)**(1/3)*constants.hbar/constants.k
+    
+
+# The definition of the Debye temperature is hbar/k_B * (6 * pi^2)^1/3 * N/V)
+# A = hbar/k_B * (6 * pi^2)^1/3
+A = 231.04 # K/(A*GPa/amu)^1/2
+BOLTZMANN = constants.physical_constants['Boltzmann constant in eV/K'][0]
+
+# check docstring bulk modulus derivative with respect to ____
+def gruneisen_parameter(
+    bulk_modulus_prime: float,
+    gruneisen_x: float
+) -> float:
+    """Calculates the gruneisen parameter
+
+    Args:
+        bulk_modulus_prime: B_0', first derivative of the bulk modulus with respect to volume
+        gruneisen_x: x, Should be between 2/3 (high temperature) and 1 (low temperature)
+
+    Returns:
+        gamma, the gruneisen parameter
+    """    
+    return (1+bulk_modulus_prime)/2 - gruneisen_x
+
+def debye_temperature(
+    volume: float,
+    eos_parameters: tuple[float],
+    mass: float,
+    gru_param: float,
+    scaling_factor: float = 0.617
+) -> float:
+    """Calculates the Debye temperature within the Debye-Gruneisen model.
+
+    Args:
+        volume: Volume of the cell
+        eos_parameters: The parameters of the equation of state: (volume_0, energy_0, bulk_modulus, bulk_modulus_prime)
+        mass: Total mass of the cell
+        gru_param: gruneisen parameter
+        scaling_factor: Scaling factor defualts to 0.617 obtained by Moruzzi et al from nonmagnetic cubic metals
+
+    Returns:
+        Debye temperature
+    """    
+    s = scaling_factor
+    volume_0 = eos_parameters[0]
+    energy_0 = eos_parameters[1]
+    bulk_modulus = eos_parameters[2]
+    bulk_modulus_prime = eos_parameters[3]
+    bulk_modulus_2prime = eos_parameters[4]
+    
+    return s * A * volume_0**(1/6) * (bulk_modulus/mass)**(1/2) * (volume_0/volume)**gru_param
+
```

Comment:
We should mention the input units: V [A^3], B [GPa], M[amu]. Convert these to V[m], B[g/(ms^2)], and M[g] in this function. Calculate A directly, as mentioned above. The 231 value in the paper always confuses me.